### ISSUE TITLE
Additional Validation methods - Gmail/Google Apps/Office 365

### ISSUE THREAD

**Issue by emeth-** [reacherhq/check-if-email-exists#937]
Gmail / Google Apps email addresses can be validated with 100% accuracy using [this method](https://blog.0day.rocks/abusing-gmail-to-get-previously-unlisted-e-mail-addresses-41544b62b2), and here's an [example implementation](https://github.com/emeth-/Email-Permutator/blob/master/background.js#L67-L86).

Office 365 email addresses are a bit trickier. There are some older enumeration options that no longer work reliably. The method outlined in [this blog post](https://www.trustedsec.com/blog/achieving-passive-user-enumeration-with-onedrive/) still works today - with one caveat. It detects valid Office 365 email addresses with 100% accuracy IF the user has opened OneDrive at least once. However, it cannot tell the difference between an invalid email, or a valid email belonging to a user who has never opened OneDrive. Still it's useful as it generates no false positives. Here's an [example implementation](https://github.com/emeth-/Email-Permutator/blob/master/js/options.js#L173-L204).

**Comment by amaurym** [comment]
This is gold. If someone wants to work on this, hit me up.

**Comment by PsypherPunk** [comment]
@amaurym, would you accept a PR for each individually? Wouldn't mind attempting at least the Gmail one.

**Comment by amaurym** [comment]
Yes, for sure! Go ahead for gmail. Please make it optional though, e.g. using a `gmail_use_api` field (defaulted to false) on CheckEmailInput.

**Comment by PsypherPunk** [comment]
I'll take a look at the Office 365 validation now, if that's okay?

**Comment by amaurym** [comment]
Yes, sure! I would also like to compare the blog's implementation versus a headless version using Outlook's password recovery page. It seems that the latter doesn't work for custom domains in MS 365 (see https://github.com/reacherhq/check-if-email-exists/issues/1185), so we might need to use this blog's API endpoint for those.

**Comment by djale1k** [comment]
Unfortunately, Google disabled the set-cookie header on this endpoint.
